piper/ratbagd.py

```diff
@@ -205,15 +219,22 @@ class Ratbagd(_RatbagdDBus):
 
     def __init__(self):
         _RatbagdDBus.__init__(self, "Manager", None)
+        result = self._get_dbus_property("Devices")
+        self._devices = [RatbagdDevice(objpath) for objpath in result]
+
+    def _on_properties_changed(self, proxy, changed_props, invalidated_props):
+        if "Devices" in changed_props.keys():
+            for prop in changed_props["Devices"]:
+                index = self._find_object_with_path(self._devices, prop)
```

Comment:
that's what `list.remove()` does for you though, it calls (iirc) `__eq__` on each device in the list. So if `__eq__` is implemented to return true if `self.objpath == that_string`, then it should work.

it's not pretty because now you have to implement `__eq__` when comparing against an object and a string, but it's doable. whether it's good python I'm not sure. Feel free to leave it as-is, `__eq__` doesn't give us that much benefit here.